tasks/libs/ciproviders/gitlab_api.py

```diff
@@ -241,3 +242,65 @@ def load_context(context):
             return [list(j.items())]
         except json.JSONDecodeError:
             raise Exit(f"Invalid context: {context}, must be a valid json, or a path to a yaml file", 1)
+
+
+def update_gitlab_config(file_path, version, patterns="", test_version=False):
+    """
+    Override variables in .gitlab-ci.yml file
+    """
+    with open(file_path) as gl:
+        file_content = gl.readlines()
+    output, images = modify_content(file_content, version, patterns, test_version)
+    with open(file_path, "w") as gl:
+        gl.writelines(output)
+    return images
+
+
+def modify_content(lines, version, patterns="", test_version=False):
+    """
+    Modify lines according to the version and patterns
+    """
+    output = []
+    modified_images = []
+    image_pattern = re.compile(r"^[ ]+CI_IMAGE_(?P<name>\w+)_(?P<id>VERSION|SUFFIX): (?P<version>.+)$")
+    version_pattern = re.compile(r"v\d+-\w+")
+    if patterns == "":
+        patterns = [""]  # Default to all images
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏Should `patterns` be None by default instead of `""` ? It seems confusing, maybe we can add type annotation such as `list[str] | None`

Reply:
I've changed the matching condition below, indeed it's better with `None`.
I didn't changed the annotations because there is no ones currently in the file, would prefer doing it in a dedicated commit.